parley/src/layout/editor.rs

```diff
@@ -424,20 +441,7 @@ where
         }
 
         self.update_layout(font_cx, layout_cx);
-        let new_start = start.saturating_add(s.len());
-        self.selection = if new_start == self.buffer.len() {
-            Selection::from_index(
-                &self.layout,
-                new_start.saturating_sub(1),
-                Affinity::Upstream,
-            )
-        } else {
-            Selection::from_index(
-                &self.layout,
-                new_start.min(self.buffer.len()),
-                Affinity::Downstream,
-            )
-        };
+        self.set_selection(self.cursor_at(start.saturating_add(s.len())).into());
```

Comment:
I'm not certain this handles affinity as I'd expect. I'd expect the cursor to be associated with the content you just pasted, not the following cluster.